### PR TITLE
Hotfix/rs convert jpg png

### LINKED ISSUE #7685: Bug: JPG/JPEG Image Upload Causes Broken Images and Errors
**Description:**
When uploading JPG or JPEG images through the CMS, the following issues occur:

1. The uploaded images are not visible or only partially visible when trying to view the actual image URL. For example:
   - Uploaded image (400px width): https://s3.amazonaws.com/digitalgov/analytics-usa-gov-over-1-point-five-million-visitors-at-3-39-pm-may-29-2024_w400.jpg (visible)
   - Full-size image: https://s3.amazonaws.com/digitalgov/title-card-navy-peo-digital-delivering-it-that-just-works-case-study-trio.jpg (not visible)

2. When adding a new blog post URL (e.g., https://digital.gov/2024/06/10/how-the-u-s-department-of-the-navy-is-delivering-it-that-just-works/) to Facebook's Debugger tool, the following issues arise:
   - An error message stating that the image is corrupted.
   - The wrong replacement image (https://digital.gov/img/digit-light.png) is used instead of the default DigitalGov title card.

**Steps to Reproduce:**
1. Upload a JPG or JPEG image through the CMS.
2. View the uploaded image in the Images feed.
3. Attempt to view the full-size image by removing the `_w400` suffix from the image URL.
4. Add the URL of a new blog post containing the uploaded image to Facebook's Debugger tool.

**Expected Behavior:**
- Uploaded JPG/JPEG images should be visible in both the 400px width and full-size versions.
- When sharing a blog post URL on Facebook, the correct image should be displayed without any errors.

**Actual Behavior:**
- Full-size versions of uploaded JPG/JPEG images are not visible or only partially visible.
- Facebook's Debugger tool reports errors and uses the wrong replacement image when sharing blog post URLs containing JPG/JPEG images.

**Additional Information:**
- The issue seems to be specific to JPG/JPEG image formats. PNG images work fine.
- The problem occurs consistently for all JPG/JPEG uploads.

### PR DESCRIPTION
**Pull request by RileySeaburg**
## Summary 

This hotfix addresses an issue where uploading JPG/JPEG images was causing errors and broken images on the website. The fix converts JPG/JPEG images to PNG format before further processing. Closes #7685.

### Preview

N/A

### Solution

- Added a new async function `convertJpgToPng` in `file-prep.js` to convert JPG/JPEG images to PNG format using the sharp library. 
- Modified the `fileTidy` function to call `convertJpgToPng` when a JPG/JPEG file is encountered. The original JPG file is deleted after conversion.
- Added error handling to log any errors that occur during the JPG to PNG conversion.
- Created a new `readme.md` file to provide an overview of the image and file upload processing functionality, usage instructions, configuration details, and dependencies.

### How To Test

1. Upload a JPG/JPEG image to the `content/uploads/_inbox` directory.
2. Run the `npx gulpupload` command.
3. Verify that the JPG/JPEG image is converted to PNG format in the `content/uploads/_working-images/to-process` directory.
4. Check the Images feed and Facebook Debugger to ensure the converted PNG image is displayed correctly without any errors.

---

### Dev Checklist

- [x] PR has correct labels 
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

**Comment by RileySeaburg** [comment]
I need to clean up some delete folder errors, but the hotfix is working.

File: config/gulp/readme.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+5. `styles.js`: Compiles SASS stylesheets, copies USWDS fonts and images, and generates SVG sprites.
+
+## Usage
+
+To use the image and file upload processing feature, follow these steps:
+
+1. Place the uploaded images and files in the designated upload directory (`content/uploads/_inbox`).
+
+2. Run the `npm run gulpupload` command to execute the entire upload process, including file preparation, image processing, and uploading to Amazon S3.
```
Comment (mejiaj) [inline config/gulp/readme.md:25]: Missing space in command name.

```suggestion
2. Run the `npm run gulp upload` command to execute the entire upload process, including file preparation, image processing, and uploading to Amazon S3.
```

File: config/gulp/readme.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+The `gulpupload` command will perform the following steps:
```
Comment (mejiaj) [inline config/gulp/readme.md:27]:
```suggestion
The `gulp upload` command will perform the following steps:
```

**Review by mejiaj — DISMISSED** [review]
> [!NOTE]  
> I only testing uploading JPG & PNG.

I was able to successfully test uploading a JPG & PNG file. The JPG was successfully converted and the PNG uploaded as-is.

---

I hope we continue to look for a workaround for JPGs or a leaner alternative. PNGs are heavier and worse quality in comparison.

**Comment by RileySeaburg** [comment]
> Note
> 
> I only testing uploading JPG & PNG.
> 
> I was able to successfully test uploading a JPG & PNG file. The JPG was successfully converted and the PNG uploaded as-is.
> 
> I hope we continue to look for a workaround for JPGs or a leaner alternative. PNGs are heavier and worse quality in comparison.

As per @nick-mon1 the action plan was to 

- [x] Add a hotfix to the image upload process 
- [ ] Determine if webp is a viable alternative.
- [ ] Replace jpg-> png code with jpg -> webp code.

Does this sound like a plan?

**Comment by RileySeaburg** [comment]
> @RileySeaburg it'd be nice to find a solution to the JPG issue either way. But I don't want to slow us down from higher priority items.
> 
> One last note, can we please remove any test images from the S3 bucket?

Done

**Review by nick-mon1 — CHANGES_REQUESTED** [review]
What I tested:

- jpg to png
- png
- static file (pptx)

Static file uploading works but the output has changed. Just making a note of the differences.


**Before**

```bash
Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
[14:42:16] Finished 'cleanInbox' after 12 ms
[14:42:16] Starting 'writeDataFile'...
[14:42:16] Finished 'writeDataFile' after 18 ms
[14:42:16] Starting 'variants'...
[14:42:16] Starting 'removeProcessedImage'...
Removing processed images
[14:42:16] Finished 'variants' after 6.6 ms
[14:42:16] Finished 'removeProcessedImage' after 6.9 ms
[14:42:16] Starting 'uploadImage'...
starting image upload
[14:42:16] Finished 'uploadImage' after 12 ms
[14:42:16] Starting 'uploadFile'...
starting file upload
file is written
[14:42:16] Uploading .....  uswds-monthly-call-april-2024.pptx
[14:42:16] Uploaded! .....  uswds-monthly-call-april-2024.pptx
[14:42:16] Finished 'uploadFile' after 341 ms
[14:42:16] Starting 'cleanup'...
[14:42:16] Finished 'cleanup' after 4.5 ms
[14:42:16] Finished 'upload' after 400 ms
```

**After**

```bash
[09:27:57] Using gulpfile ~/Documents/gsa/digitalgov.gov/gulpfile.js
[09:27:57] Starting 'upload'...
[09:27:57] Starting 'fileTidy'...
Moving file from content/uploads/_inbox/s3-test-uswds-monthly-call-may-2024.pptx to content/uploads/_working-files/to-process/s3-test-uswds-monthly-call-may-2024.pptx
[09:27:57] Finished 'fileTidy' after 2.58 ms
[09:27:57] Starting 'cleanInbox'...
[09:27:57] Finished 'cleanInbox' after 8.86 ms
[09:27:57] Starting 'writeDataFile'...
[09:27:57] Finished 'writeDataFile' after 29 ms
[09:27:57] Starting 'processImages'...
[09:27:57] Finished 'processImages' after 308 μs
[09:27:57] Starting 'removeProcessedImage'...
Removing processed images
No images to process
Error processing file: [undefined]. ENOENT: no such file or directory, scandir './content/uploads/_working-images/to-process'
[09:27:57] Finished 'removeProcessedImage' after 2.19 ms
[09:27:57] Starting 'uploadImage'...
[09:27:57] Finished 'uploadImage' after 12 ms
[09:27:57] Starting 'uploadFile'...
file is written
(node:6841) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
[09:28:09] [create] static/s3-test-uswds-monthly-call-may-2024.pptx
[09:28:09] Finished 'uploadFile' after 12 s
[09:28:09] Starting 'cleanup'...
cleanup
[09:28:09] Finished 'cleanup' after 5.93 ms
[09:28:09] Finished 'upload' after 12 s
```

File: config/gulp/readme.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+5. `styles.js`: Compiles SASS stylesheets, copies USWDS fonts and images, and generates SVG sprites.
+
+## Usage
+
+To use the image and file upload processing feature, follow these steps:
+
+1. Place the uploaded images and files in the designated upload directory (`content/uploads/_inbox`).
+
+2. Run the `npm run gulp upload` command to execute the entire upload process, including file preparation, image processing, and uploading to Amazon S3.
```
[the remark this answers is not in the document]
Comment (RileySeaburg) [reply config/gulp/readme.md:25]: Should add the script to package.json

File: config/gulp/file-prep.js [hunk 4]
```diff
@@ -53,34 +78,54 @@ function fileTidy(done) {
...
         if (filetype === "image") createDir(paths[filetype].processed, 3);
-        // copies uploaded file to /to-process with new normalized name
-        fs.renameSync(
-          `${paths.uploads}/${file}`,
-          `${paths[filetype].toProcess}/${newfileName}`
-        );
+        // copies uploaded file to /to-process with new normalized name 
+        // convert jpg to png
+        if (filetype === "image" && fileExt === ".jpg" || fileExt === ".jpeg") {
+         let result =  await convertJpgToPng(path.join(paths.uploads, file)).catch
```
[the remark this answers is not in the document]
Comment (RileySeaburg) [reply config/gulp/file-prep.js]: All set

**Comment by RileySeaburg** [comment]
All set on this

**Review by nick-mon1 — APPROVED** [review]

**Review by mejiaj — DISMISSED** [review]

**Comment by RileySeaburg** [comment]
New logs associated with this PR

image

**Comment by RileySeaburg** [comment]
Closed merged into #7671

**Pull request closed without merging**